service/worker/batcher/activities.go

```diff
@@ -308,3 +332,98 @@ func isDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+func getResetEventIDByType(ctx context.Context,
+	resetType enumspb.ResetType,
+	namespaceStr string,
+	workflowExecution *commonpb.WorkflowExecution,
+	frontendClient workflowservice.WorkflowServiceClient,
+	logger log.Logger) (int64, error) {
+	switch resetType {
+	case enumspb.RESET_TYPE_FIRST_WORKFLOW_TASK:
+		return getFirstWorkflowTaskEventID(ctx, resetType, namespaceStr, workflowExecution, frontendClient, logger)
+	case enumspb.RESET_TYPE_LAST_WORKFLOW_TASK:
+		return getLastWorkflowTaskEventID(ctx, resetType, namespaceStr, workflowExecution, frontendClient, logger)
+	default:
+		errorMsg := fmt.Sprintf("provided reset type (%v) is not supported.", resetType)
+		return 0, serviceerror.NewInvalidArgument(errorMsg)
+	}
+}
+
+func getLastWorkflowTaskEventID(ctx context.Context,
+	resetType enumspb.ResetType,
+	namespaceStr string,
+	workflowExecution *commonpb.WorkflowExecution,
+	frontendClient workflowservice.WorkflowServiceClient,
+	logger log.Logger) (workflowTaskEventID int64, err error) {
+	req := &workflowservice.GetWorkflowExecutionHistoryReverseRequest{
+		Namespace:       namespaceStr,
+		Execution:       workflowExecution,
+		MaximumPageSize: 1000,
+		NextPageToken:   nil,
+	}
+	for {
+		resp, err := frontendClient.GetWorkflowExecutionHistoryReverse(ctx, req)
+		if err != nil {
+			logger.Error("failed to run GetWorkflowExecutionHistoryReverse")
+			return 0, errors.New("failed to get workflow execution history")
+		}
+		for _, e := range resp.GetHistory().GetEvents() {
+			if e.GetEventType() == enumspb.EVENT_TYPE_WORKFLOW_TASK_COMPLETED {
+				workflowTaskEventID = e.GetWorkflowTaskCompletedEventAttributes().GetScheduledEventId()
```

Comment:
1. if reset to the scheduled event, need +1 to eventID, because existing reset logic still needs the scheduled event.
2. why reset to the scheduled event instead of the completed event?